src/nspawn/nspawn-network.c

```diff
@@ -168,12 +168,36 @@ static int add_veth(
         return 0;
 }
 
+static char *shorten_ifname(char *ifname) {
+        unsigned char hash = 0xff;
+
+        assert(ifname);
+
+        if (strlen(ifname) >= IFNAMSIZ) {
+                _cleanup_free_ char *old_ifname = strdup(ifname);
+
+                /* Create a simple 8-bit hash of an interface name */
+                for (char *pos = ifname; *pos != 0; pos++)
+                        hash = ((hash << 1) | (hash >> 7)) ^ *pos;
+
+                /* Shorten the string and insert hash */
+                snprintf(ifname + IFNAMSIZ - 3, 3, "%02x", hash);
+
+                /* Log the incident to make it more discoverable */
+                log_warning("Interface '%s' has been renamed to '%s' to fit length constraints.",
```

Comment:
hmm, maybe downgrade this to log_notice(), i.e. higher than the usual logging, but still not really a problem

Reply:
In my first attempt I had `log_notice()` but looking at the code it was mainly used for amending warnings with additional information. That's why I went for `log_warning()` instead. But okay, I'll change it. Makes sense to me.

Reply:
Fixed in next push